### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by rhamzeh**
## Description

Go:
- Set target go to 1.20
- Update dependencies

JS:
- Fix types
- Update dependencies
- Release v0.2.0-beta.8

Note: Go test are expected to fail as it does not support nesting yet

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/go/go.mod [hunk 1]
```diff
@@ -1,13 +1,13 @@
 module github.com/openfga/language/pkg/go
 
-go 1.21.5
+go 1.20
```
Comment (sergiught) [inline pkg/go/go.mod:3]: What's causing is to downgrade?

**Review by ewanharris — APPROVED** [review]

File: pkg/go/go.mod (hunk 1 above)
Comment (rhamzeh) [reply pkg/go/go.mod:3]: A third party faced an issue with us using go1.21, we're gonna do a latest - 1 approach, so support the latest minor + the one before it. So while current is go1.21, we'll support go1.20+

See here: https://github.com/lxc/incus/issues/315#issuecomment-1863381991

Comment (sergiught) [reply pkg/go/go.mod:3]: Understood:) thanks for the reference. Should we then publicly follow the same release policy as Go itself and state it publicly within all our customer facing SDKs/Tools https://go.dev/doc/devel/release#policy?

Comment (rhamzeh) [reply pkg/go/go.mod:3]: I dunno if we want to publicly commit to it yet versus a "best effort". At least for now

**Review by sergiught — APPROVED** [review]

**Pull request merged**